src/env/node/git/git.ts

```diff
@@ -160,6 +167,14 @@ function getStdinUniqueKey(): number {
 type ExitCodeOnlyGitCommandOptions = GitCommandOptions & { exitCodeOnly: true };
 export type PushForceOptions = { withLease: true; ifIncludes?: boolean } | { withLease: false; ifIncludes?: never };
 
+const resetErrorAndReason = [
+	[unmergedChanges, ResetErrorReason.UnmergedChanges],
+	[ambiguousArgument, ResetErrorReason.AmbiguousArgument],
+	[entryNotUpToDate, ResetErrorReason.EntryNotUpToDate],
+	[changesWouldBeOverwritten, ResetErrorReason.LocalChangesWouldBeOverwritten],
+	[refLocked, ResetErrorReason.RefLocked],
```

Comment:
The ones on the left should be `GitErrors.unmergedChanges` etc.

Also should set a type for `resetErrorAndReason` explicitly because the auto-type is not as specific as we want.